### DISCUSSION TITLE
Where is the __except identifier declared?

### DISCUSSION THREAD

**Discussion by TheShermanTanker** [msys2/MINGW-packages#11609] [Q&A]
To my knowledge gcc doesn't support the __try __except that Visual C++ does, but while working on adding compiler support a codebase I'm getting an error for only __try not being defined in the scope, __except seems to be resolved perfectly fine, without any issues. I haven't included excpt.h anywhere so far, and am not using the gcc equivalents __try1 and __except1, so where is __except being defined?

**Comment by revelator** [comment]
i think it was added to the mingw-w64 api as wrappers at some time as i stumbled upon them by accident but i cannot remember exactly where in the sources.

**Reply by TheShermanTanker** [reply]
darn, that's a bummer. No worries though

**Reply by revelator** [reply]
aye can be a bit confusing if the user is unaware, though i think it was done as a workaround for some other exception specific tasks mingw users had at the time.

**Reply by revelator** [reply]
i was wrong they are actually defined in libstdc++ headers which also makes using stuff like libseh a pain in the neck to use with 32 bit :S.
__try and __catch are defined in /mingw*/include/c++/<version>/<arch>/bits/exception_defines.h and __except seems to be defined in the mingw abi.
